### DISCUSSION TITLE
Improve History Page Speed

### DISCUSSION THREAD

**Discussion by sim-san** [home-assistant/frontend#11488] [Other feature requests]
Hi,
the history page is slow and not really interactive. 
Currently, there are two new custom integration that tries to create a smooth interactive experience to view the history of your entities.
This is achieved by:

> The card uses asynchronous stream caching and adaptive data decimation to hide the high latency of HA's history database accesses and tries to make it into a smooth interactive experience.

- [history-explorer-card](https://github.com/alexarch21/history-explorer-card)
- [lovelace-plotly-graph-card](https://github.com/dbuezas/lovelace-plotly-graph-card)

Is it possible to use these technologies also in the offical history Page ?

**Comment by Misiu** [comment]
Or use https://github.com/leeoniya/uPlot as mentioned here https://github.com/home-assistant/frontend/discussions/11251

**Comment by sim-san** [comment]
Is there a plan to work on the History Page Speed ?

**Comment by RoboMagus** [comment]
Not reloading the entire data set that is requested upon a selection update would help massively.
With the (Awesome) addition of selecting multiple entities to show in the history tab, when adding / removing an entity it seems that the entire dataset is updated where a partial update would suffice.
